src/trio/_core/_tests/test_run.py

```diff
@@ -123,10 +124,11 @@ async def test_nursery_warn_use_async_with() -> None:
 async def test_nursery_main_block_error_basic() -> None:
     exc = ValueError("whoops")
 
-    with pytest.raises(ValueError) as excinfo:
+    with pytest.raises(ExpectedExceptionGroup(ValueError)) as excinfo:
         async with _core.open_nursery():
             raise exc
-    assert excinfo.value is exc
+    assert len(excinfo.value.exceptions) == 1
```

Comment:
I noticed you removed this kind of assertation (checking that the `ExceptionGroup` has *exactly* one exceptions) elsewhere. Is this redundent? Or should `ExpectedExceptionGroup` has a `strict` kwarg that means it doesn't take other exceptions (in a specific ordering? not sure about the impl...)?

Reply:
Yeah `ExpectedExceptionGroup` by default checks that *exactly* the exceptions you specify get raised, no more, no less. A kwarg that changes that behaviour would make sense though